src/Stratis.Bitcoin/Base/ChainHeadersBehavior.cs

```diff
@@ -389,10 +395,24 @@ public void TrySync()
             this.logger.LogTrace("(-)");
         }
 
-        private ChainedBlock GetPendingTipOrChainTip()
+        /// <summary>
+        /// Determines if the peer's headers are synced with ours.
+        /// </summary>
+        /// <remarks>
+        /// It is possible that peer is in IBD even though he has all the headers so we can't assume with 100% certainty that peer is fully synced.
+        /// </remarks>
+        public bool IsSynced()
+        {
+            if (this.pendingTip == null)
+                return false;
+
+            return (this.pendingTip.Height >= this.chainState.ConsensusTip.Height &&
+                    this.pendingTip.FindAncestorOrSelf(this.chainState.ConsensusTip) != null);
```

Comment:
why this condition?
would it not be better to check ChainWork?

Reply:
Good idea. Fixed.